staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -233,6 +242,28 @@ func NewCloud(configReader io.Reader) (cloudprovider.Interface, error) {
 		return nil, err
 	}
 
+	az := &Cloud{
+		nodeZones:          map[string]sets.String{},
+		nodeResourceGroups: map[string]string{},
+		unmanagedNodes:     sets.NewString(),
+		routeCIDRs:         map[string]string{},
+	}
+	err = az.initializeCloudFromConfig(config, false)
+	if err != nil {
+		return nil, err
+	}
+
+	return az, nil
+}
+
+// initializeCloudFromConfig initializes the Cloud from config.
+func (az *Cloud) initializeCloudFromConfig(config *Config, fromSecret bool) error {
```

Comment:
can you also make this function as public, that would be more useful as a lib, other part LGTM.

Reply:
done.